### ISSUE TITLE
[Feedback] custom cf does not require `dependsOn`

### ISSUE THREAD

**Issue by cyrfer** [aws-amplify/docs#2101]
**Page**: [`/cli/usage/customcf`](https://docs.amplify.aws/cli/usage/customcf)

**Feedback**:

I don't think this is true:
> you would need to add the dependsOn block in the backend-config.json 

I have had success using only the `backend/[category]/[instance]/parameters.json` file to pass output from one stack as a parameter into the stack in the folder hierarchy. See here for more info:
https://github.com/aws-amplify/amplify-cli/issues/4006#issuecomment-623057081

**Comment by edwardfoyle** [comment]
@cyrfer You need to update the dependsOn block to ensure that dependencies are synthesized before dependent resources. If your resources happen to be synthesized in a topological order, then it would work but otherwise would fail.

**Comment by cyrfer** [comment]
This is really old, so I may not remember all the issues.

@edwardfoyle 
In my head, If I’m passing a parameter to a stack that is sourced from an output of another stack, the other stack must already exist, and therefore there is no need to use dependsOn.

**Comment by cyrfer** [comment]
@edwardfoyle I bumped into the issue you anticipated. Despite my ideals that dependsOn is over-specified, the reality is developers must use it, especially when creating a new environment for the first time after linking sub-stacks with `Fn::GetAtt` in `category/resource/parameters.json`.

I think `dependsOn` in `backend/backend-config.json` should be described completely.
https://github.com/aws-amplify/docs/issues/2437

Here is a sample of how I used it, helped by the subtle comments in the code blocks on page https://docs.amplify.aws/cli/usage/customcf
```json
"dependsOn": [
  {"category": "mycategoryfoldername", "resourceName": "visibleFromAmplifyStatus", "attributes": ["OneOfMyOutputs"]}
],
```
